src/SimpleDTO.php

```diff
@@ -399,15 +398,26 @@ private function isArrayable(mixed $value): bool
 
     private function formatArrayableValue(mixed $value): array
     {
-        if (is_array($value)) {
-            return $value;
-        }
+        return match (true) {
+            is_array($value) => $value,
+            $value instanceof Collection => $value->toArray(),
+            $value instanceof Model => $value->toArray(),
+            $value instanceof SimpleDTO => $this->transformDTOToArray($value),
+            is_object($value) => (array) $value,
```

Comment:
This has broken Enums. A UnitEnum is now `['name' => 'Foo']` and a BackedEnum is now `['name' => Foo, 'value' => 'Bar']`